Sources/SwiftInspectorCommands/Tests/TypesCommandSpec.swift

```diff
@@ -0,0 +1,125 @@
+// Created by Tyler Hedrick on 8/13/20.
+//
+// Copyright (c) 2020 Tyler Hedrick
+//
+// Distributed under the MIT License
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in all
+// copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+// SOFTWARE.
+
+import Nimble
+import Quick
+import Foundation
+
+@testable import SwiftInspectorCore
+
+final class TypesCommandSpec: QuickSpec {
+  override func spec() {
+    var fileURL: URL!
+
+    afterEach {
+      guard let fileURL = fileURL else {
+        return
+      }
+      try? Temporary.removeItem(at: fileURL)
+    }
+
+
+    describe("run") {
+      context("with no arguments") {
+        it("fails") {
+          let result = try? TestTask.run(withArguments: ["types"])
+          expect(result?.didFail) == true
+        }
+      }
+
+      context("when path is invalid") {
+        it("fails when empty") {
+          let result = try? TestTypesCommandTask.run(path: "")
+          expect(result?.didFail) == true
+        }
+
+        it("fails when it doesn't exist") {
+          let result = try? TestTypesCommandTask.run(path: "/fake/path")
+          expect(result?.didFail) == true
+        }
+      }
+
+      context("when path is valid") {
+        it("succeeds") {
+          fileURL = try? Temporary.makeFile(content: "struct Foo { }")
+          let path = fileURL?.path ?? ""
+          let result = try? TestTypesCommandTask.run(path: path)
+          expect(result?.didSucceed) == true
+        }
+
+        it("runs and outputs the result without comment") {
+          fileURL = try? Temporary.makeFile(content: """
+                                                   // This is a comment
+                                                   struct Foo { }
+                                                   """)
+          let path = fileURL?.path ?? ""
+          let result = try? TestTypesCommandTask.run(path: path)
+          expect(result?.outputMessage).to(contain("Foo,struct"))
```

Comment:
We can refactor these 2 tests to put the common file in a `beforeEach` so this looks like:

```swift
var result: TaskStatus!

beforeEach {
  fileURL = try? Temporary.makeFile(content: "struct Foo { }")
  let path = fileURL?.path ?? ""
  result = try? TestTypesCommandTask.run(path: path)
}

it("succeeds") {
  expect(result?.didSucceed) == true
}

it("outputs the type name and Type") {
  expect(result?.outputMessage).to(contain("Foo,struct"))
}
```

Reply:
@fdiaz i'm a bit confused, these aren't the same file. One contains a comment while the other does not.

Reply:
No worries, this was mostly a nit so let's keep it as is 😉